generators/advanced/templates/root/_README.md

```diff
@@ -0,0 +1,166 @@
+
+### __routes.api.v2__
+The api v2 sub-component in charge of exposing the `/v2/message/{id}` endpoint. It is capable of dealing with the defualt http errors thanks to the module [error-handler-module](https://www.npmjs.com/package/error-handler-module) (e.g. 404 NOT_FOUND_ERROR).
+
+### __controller__
+Is the orchestrator of the different components and subcomponents of the system, handles the core logic of the Hexagon Architecture.
+
+### __controller.api.v1__
+Implements the v1 business logic needed to manage the api v1 routes requests. It retrieves a given message from the store and manipulates its payload in order to give back to the api v1 request a response containing:
+- `id`: original message id.
+- `text`: lowercase original message text.
+
+### __controller.api.v2__
+Implements the v2 business logic needed to manage the api v2 routes requests. It retrieves a given message from the store and manipulates its payload in order to give back to the api v2 request a response containing:
+- `id`: original message id.
+- `text`: uppercase original message text.
+- `receptionTimestamp`: ISO string timestamp of the original message reception time.
+
+### __controller.bus__
+Implements the business logic needed to manage the messages coming from the rabbitmq bus. It is in charge of storing the original payload messages along with the reception timestamp into the monogodb store.
+
+### __store__
+Manages the connection with the mongodb database exposing storing and retrieving functionalities of messages which in turn will be exploited by the controllers.
+
+
+# CONFIG
+
+The configuration is scattered across different files and mixed according to the service run mode as said above.
+The intesting bit is the fact that is stored in js objects and is definable per-component and per-sub-component. 
+
+E.g. __routes__ and __routes.admin__ [default configuration](config/default.js):
+```js
+{
+    routes: {
+		admin: {
+			swaggerValidator: {
+				apiDocEndpoint: '/__/docs/api',
+				validateRequests: true,
+				validateResponses: true,
+				validationEndpoint: '/test',
+				format: 'yaml',
+				yaml: {
+					file: './docs/syncapi.yaml',
+				},
+			},
+		},
+    },
+}
+```
+
+Note that the configuration varies according to the different run modes. Particularly the rabbitmq one for the systemic-rabbitmq component coming as an npm library:
+- __Default__: according the the business logic of our service the default config is needed only for the service to be subscribed to a given queue.
+- __Test__: in order to test the service business logic during the test phase is the very same service the one publishing the messages that is going to consume. More info in the test section.
+
+
+# DOCS
+The _/docs_ folder holds the synchronous api documentation for the endpoints exposed by the __routes__ component.
```

Comment:
Should we mention here that it's also possible to use `yaml_jsdoc` format?

I have to improve the `swagger-endpoint-validator`, and it'll appear in its README.  So if don't want to add complexity, then we can leave it as it is now.

Reply:
I believe that is something more related to the docs of the swagger-endpoint-validator npm module, in the end here we are just saying that we are using the module but we should not explain too much in details how the module works, that, again, is responsibility of the module readme.